pegasus/sites.v3/code.org/public/js/workshop-search.js

```diff
@@ -23,83 +25,89 @@ function initializeMap() {
 }
 
 function loadWorkshops() {
-  var params = [];
+  markerClusterer = new MarkerClusterer(gmap, [],
+    {imagePath: 'https://developers.google.com/maps/documentation/javascript/examples/markerclusterer/m'}
+  );
 
   $.get('/dashboardapi/v1/pd/k5workshops').done(function (data) {
     processPdWorkshops(data);
-  });
+  }).always(completeProcessingPdWorkshops);
 
-  $.post('/forms/ProfessionalDevelopmentWorkshop/query', $.param(params), function (response) {
-    var results = JSON.parse(response);
-
-    $.each(results.response.docs, function (index, workshop) {
-      if (typeof workshop.location_p !== 'undefined') {
-        var location = workshop.location_p.split(',');
-        var latLng = new google.maps.LatLng(location[0], location[1]);
-
-        // This hash will round the latLngs to 6 decimals.
-        var hash = latLng.toUrlValue();
-
-        var infoWindowContent = '';
-
-        if (typeof markers[hash] === 'undefined') {
-          infoWindowContent = compileHtml(workshop, true);
-          markers[hash] = new google.maps.Marker({
-            position: latLng,
-            map: gmap,
-            title: workshop.location_name_p,
-            infoWindowContent: infoWindowContent
-          });
-          google.maps.event.addListener(markers[hash], "click",
-            function (e) {
-              infoWindow.setContent(this.get('infoWindowContent'));
-              infoWindow.open(gmap, this);
-            });
-        } else {
-          infoWindowContent = compileHtml(workshop, false);
-          // Extend existing marker.
-          markers[hash].infoWindowContent += infoWindowContent;
-        }
+  $.post('/forms/ProfessionalDevelopmentWorkshop/query', function (data) {
+    processLegacyWorkshops(data);
+  }).always(completeProcessingLegacyWorkshops);
+}
+
+function processLegacyWorkshops(response) {
+  var results = JSON.parse(response);
+
+  $.each(results.response.docs, function (index, workshop) {
+    if (typeof workshop.location_p !== 'undefined') {
+      var location = workshop.location_p.split(',');
+      var latLng = new google.maps.LatLng(location[0], location[1]);
+
+      // This hash will round the latLngs to 6 decimals.
+      var hash = latLng.toUrlValue();
+
+      var infoWindowContent = '';
+
+      if (typeof markersByLocation[hash] === 'undefined') {
```

Comment:
Good question. I didn't modify [that line](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/code.org/public/js/workshop-search.js#L36). I'll update it too...